batcher/metrics/metrics.go

```diff
@@ -85,10 +85,10 @@ func (bld *MetricsBatchBuilder) Build(input [][]byte) [][]byte {
 func applyTime(
 	res *transit.DynamicMonitoredResource,
 	svc *transit.DynamicMonitoredService,
-	ts milliseconds.MillisecondTimestamp,
+	ts *milliseconds.MillisecondTimestamp,
 ) {
-	if ts.IsZero() {
-		ts = milliseconds.MillisecondTimestamp{Time: time.Now()}
+	if ts == nil || ts.IsZero() {
+		ts = &milliseconds.MillisecondTimestamp{Time: time.Now()}
```

Comment:
I don't see how this is going to work.  milliseconds.MillisecondTimestamp() is going to return a result which is probably in a machine register, and you won't be able to take the address of that value.

Reply:
It works in Go.